firmware/watchguard_sysa_dl.ksy

```diff
@@ -0,0 +1,137 @@
+meta:
+  id: watchguard_sysa_dl
+  endian: be
+  title: Watchguard Fireware web update (Sysa-dl) file format
+  license: MIT
+
+seq:
+  - id: fw_header
+    type: fw_header
+    size: 24
+
+  - id: sections
+    type: sections
+    size: fw_header.file_size
+
+types:
+  fw_header:
+    seq:
+      - id: md5sum
+        size: 0x10
+
+      - id: file_size
+        type: u4
+
+      - id: magic
+        type: u4
+        enum: magic
+
+  sections:
+    seq:
+      - id: section
+        type: section
+        repeat: eos
+
+  section:
+    seq:
+      - id: head
+        type: head
+
+      - id: data
+        size: head.data_size
+        type:
+          switch-on: head.name.content
+          cases:
+            '"REBOOT"': reboot_data
+            '"info"': info_data
+            '"HMAC"': hmac_data
+            '"WGPKG"': wgpkg_data
+
+
+  name:
+    seq:
+      - id: type2_magic
+        type: u1
+        if: is_type2
+
+      - id: content
+        type: strz
+        size: 'is_type2 ? 0xf : 0x10'
+        encoding: ASCII
+
+    instances:
+      type2:
+        pos: 0
+        type: u1
+
+      is_type2:
+        value: type2 == 0x7f
+
+  head:
+    seq:
+      - id: name
+        type: name
+        size: 0x10
+
+      - id: unknown_data
+        size: 24
+        if: name.is_type2
+
+      - id: data_size
+        type: u4
+
+      - id: md5sum
+        size: 0x10
+        if: name.is_type2
+
+  reboot_data:
+    seq:
+      - id: perm
+        type: encoded_perm
+        size: 0x8
+
+  encoded_perm:
+    seq:
+      - id: unknown_dword1
+        type: u4
+      - id: unknown_dword2
+        type: u4
+
+  info_data:
+    seq:
+      - id: info
+        type: strz
+        encoding: ASCII
+        size-eos: true
+
+  hmac_data:
+    seq:
+      - id: sha1_sign #key: etaonrishdlcupfm
+        size: 20
+
+  wgpkg_data:
+    seq:
+      - id: meta_info
+        type: strz
+        encoding: ASCII
+        terminator: 0
```

Comment:
Using both `type: strz` and `terminator` is [discouraged](https://github.com/kaitai-io/ksy_schema/pull/6#discussion_r356883541) (and BTW it's disallowed in [ksy_schema](https://github.com/kaitai-io/ksy_schema) since https://github.com/kaitai-io/ksy_schema/commit/1fc952f1e4bbb01556e75b46f44f83e6162e9841). I suggest simply omitting the `terminator: 0`:

```suggestion
      - id: meta_info
        type: strz
        encoding: ASCII
```

To be clear, `type: strz` + `terminator` is still accepted by the compiler, just not recommended.